src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -5822,6 +5864,316 @@ .maxstack  2
                 """);
         }
 
+        [CombinatorialData]
+        [ConditionalTheory(typeof(CoreClrOnly))]
```

Comment:
Why does this need to be `CoreClrOnly`?

Reply:
Can we execute tests targeting TargetFramework.Net70 or .Net80 on desktop?
```
System.BadImageFormatException : Cannot load a reference assembly for execution.
```
Are we missing any test coverage by associating these tests with CoreClrOnly?

Reply:
This is the same thing that I asked you to file a followup for last week in https://github.com/dotnet/roslyn/pull/68930#discussion_r1268352244, and offline as well. We do not need to execute the resulting binary on all platforms, but we should make sure there are no differences between the framework compiler and the core compiler in what code they emit given a set of references, as this is a realistic scenario anyone using VS will encounter.